test/CAPI/ir.c

```diff
@@ -103,6 +103,26 @@ int testHWTypes() {
   if (!hwTypeIsAInOut(io8type))
     return 4;
 
+  MlirStringRef scope = mlirStringRefCreateFromCString("myscope");
```

Comment:
I realize that few of the other tests do this, but maybe add some stdout and filecheck tests?

Reply:
I'm not sure what you mean exactly. This is how CIRCT and upstream MLIR test the C API. What would I print to STDOUT from the C API? `MlirType` is just a wrapper around a pointer. All that stuff to print types and attributes nicely comes from MLIR's C++ libraries. The Python also (indirectly) covers what these print as, but again that's with the help of the C++ libraries.

Reply:
I was thinking that you should print the strings and use filecheck to ensure correctness rather than C `if`s. If this is the standard for C API tests, I withdrawal my suggestion.

Reply:
I guess for the strings it would make sense to print and filecheck them... I can take a look at that as a follow up.

Reply:
Don't bother. If this is the way the rest of the c api tests work, better to stick with that.